src/Thumbnail.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import classSet from 'classnames';
+import BootstrapMixin from './BootstrapMixin';
+
+const Thumbnail = React.createClass({
+  mixins: [BootstrapMixin],
+
+  getDefaultProps() {
+    return {
+      bsClass: 'thumbnail'
+    };
+  },
+
+  render() {
+    let classes = this.getBsClassSet();
+
+    if(this.props.href) {
+      return (
+        <a {...this.props} href={this.props.href} className={classSet(this.props.className, classes)}>
+          <img src={this.props.src} alt={this.props.alt} />
+        </a>
+      );
+    }
+    else {
+      return (
+        <div {...this.props} className={classSet(this.props.className, classes)}>
+          <img src={this.props.src} alt={this.props.alt} />
+          <div className="caption">
+            {this.props.children}
```

Comment:
I will change this to ensure that if there are not children, the caption div will not be created.